### PR TITLE
chore: bump docusaurus, use docusaurus/preset-classic

### PR DESCRIPTION
**Pull request by Josh-Cena**
## PR Checklist

-   [ ] Addresses an existing open issue: fixes #000
-   [ ] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
-   [ ] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

## Overview

Since we are considering adding a blog (https://github.com/typescript-eslint/typescript-eslint/issues/4957), we will be enjoying all features that preset-classic provides by then. It will be sensible for us to use the preset instead, so we don't need to manage so many dependencies individually.

There are no breaking changes that impact us, but I've made a few little refactors.

### CODE REVIEW

File: packages/website/docusaurus.config.js [hunk 1]
```diff
@@ -21,44 +21,40 @@ const config = {
...
-        routeBasePath: 'rules',
-        editUrl: `${githubUrl}/edit/main/packages/website/`,
-        beforeDefaultRemarkPlugins,
-        remarkPlugins,
-        exclude: ['TEMPLATE.md'],
-        breadcrumbs: false,
-      },
+      'classic',
+      /** @type {import('@docusaurus/preset-classic').Options} */
+      ({
```
Comment (Josh-Cena) [inline packages/website/docusaurus.config.js:28]: JSDoc only works if you wrap the object literal with brackets.

File: packages/website/src/css/custom.css [hunk 2]
```diff
@@ -60,17 +61,7 @@ html[data-theme='dark']:root {
   scrollbar-color: #454a4d #202324;
 
   --docsearch-muted-color: #aaa;
-}
-
-.docusaurus-highlight-code-line {
```
Comment (Josh-Cena) [inline packages/website/src/css/custom.css:65]: This is a breaking change: the class name has been removed in Docusaurus. We are using CSS vars instead.

File: packages/website/src/css/prism.css [hunk 3]
```diff
@@ -56,8 +56,8 @@ html[data-theme='dark'] {
   --token-color-highlight: rgba(247, 235, 198, 0.2);
 }
 
-div[class*='codeBlockContainer'],
-div[class*='codeBlockContainer'] pre {
+.theme-code-block,
```
Comment (Josh-Cena) [inline packages/website/src/css/prism.css:59]: This is a more stable class name and is recommended.

File: packages/website/src/theme/prism-include-languages.js [hunk 4]
```diff
@@ -1,27 +1,16 @@
...
 
-    require(`../prism/language/jsonc`);
-    delete window.Prism;
-  }
-};
-
-export default prismIncludeLanguages;
+  require(`../prism/language/jsonc`);
+  delete globalThis.Prism;
+}
```
Comment (Josh-Cena) [inline packages/website/src/theme/prism-include-languages.js:16]: Backporting changes from upstream. By using `globalThis`, we can now support server-side rendering of custom languages as well.

**Comment by Josh-Cena** [comment]
I _think_ the reason why Cypress tests always fail on version bump PRs is because the cache is invalidated, causing `yarn start` to take longer than necessary.

File: packages/website/src/css/prism.css [hunk 5]
```diff
@@ -56,10 +56,10 @@ html[data-theme='dark'] {
   --token-color-highlight: rgba(247, 235, 198, 0.2);
 }
 
-div[class*='codeBlockContainer'],
-div[class*='codeBlockContainer'] pre {
-  color: var(--token-color);
-  background-color: var(--token-background);
+.theme-code-block,
+.theme-code-block pre {
+  color: var(--token-color) !important;
```
Comment (Josh-Cena) [inline packages/website/src/css/prism.css:61]: This is to work around a Docusaurus bug: https://github.com/facebook/docusaurus/issues/3678

File: packages/website/src/clientModules.js [hunk 6]
```diff
@@ -1,3 +1,5 @@
-if (typeof window !== 'undefined') {
+import ExecutionEnvironment from '@docusaurus/ExecutionEnvironment';
+
+if (ExecutionEnvironment.canUseDOM) {
```
Comment (JoshuaKGoldberg) [inline packages/website/src/clientModules.js:3]: Ha, I remember bumping into this discussion! Glad it's resolved 🙂

**Review by JoshuaKGoldberg — APPROVED** [review]
Fantastic, thanks so much @Josh-Cena (as always)! 👏

File: packages/website/src/clientModules.js (hunk 6 above)
Comment (Josh-Cena) [reply packages/website/src/clientModules.js:3]: Except that `ExecutionEnvironment.canUseDOM` does exactly the same thing as `typeof window !== 'undefined'` (except doing a few more checks) 🤦‍♂️

It's now officially made its way to the docs, now that client modules are finalized and published: https://docusaurus.io/docs/advanced/client#:~:text=remember%20to%20check%20the%20execution%20environment

**Pull request merged**